stdlib/public/core/UnsafePointer.swift

```diff
@@ -205,44 +204,133 @@
+extension UnsafePointer: Sendable where Pointee: ~Copyable {}
+
+@_preInverseGenerics
+extension UnsafePointer: _Pointer where Pointee: ~Copyable {
+  /// A type that represents the distance between two pointers.
+  public typealias Distance = Int
+}
 
+@_preInverseGenerics
+extension UnsafePointer: Equatable where Pointee: ~Copyable {}
+
+@_preInverseGenerics
+extension UnsafePointer: Hashable where Pointee: ~Copyable {
+  // Note: This explicit `hashValue` applies @_preInverseGenerics to emulate the
+  // original (pre-6.0) compiler-synthesized version.
+  @_preInverseGenerics
+  public var hashValue: Int {
+    _hashValue(for: self)
+  }
+}
+@_preInverseGenerics
+extension UnsafePointer: Comparable where Pointee: ~Copyable {}
+
+@_preInverseGenerics
+extension UnsafePointer: Strideable where Pointee: ~Copyable {}
+
+#if !$Embedded
+@_preInverseGenerics
+extension UnsafePointer: CustomDebugStringConvertible
+where Pointee: ~Copyable {}
+#endif
+
+#if SWIFT_ENABLE_REFLECTION
+@_preInverseGenerics
+extension UnsafePointer: CustomReflectable where Pointee: ~Copyable {}
+#endif
+
+extension UnsafePointer where Pointee: ~Copyable {
   /// Deallocates the memory block previously allocated at this pointer.
   ///
-  /// This pointer must be a pointer to the start of a previously allocated memory 
-  /// block. The memory must not be initialized or `Pointee` must be a trivial type.
+  /// This pointer must be a pointer to the start of a previously allocated
+  /// memory block. The memory must not be initialized or `Pointee` must be a
+  /// trivial type.
   @inlinable
+  @_preInverseGenerics
   public func deallocate() {
     // Passing zero alignment to the runtime forces "aligned
     // deallocation". Since allocation via `UnsafeMutable[Raw][Buffer]Pointer`
     // always uses the "aligned allocation" path, this ensures that the
     // runtime's allocation and deallocation paths are compatible.
     Builtin.deallocRaw(_rawValue, (-1)._builtinWordValue, (0)._builtinWordValue)
   }
+}
 
+extension UnsafePointer where Pointee: ~Copyable {
   /// Accesses the instance referenced by this pointer.
   ///
   /// When reading from the `pointee` property, the instance referenced by
   /// this pointer must already be initialized.
-  @inlinable // unsafe-performance
+  @_alwaysEmitIntoClient
   public var pointee: Pointee {
     @_transparent unsafeAddress {
       return self
     }
   }
+}
+
+extension UnsafePointer {
+  // This preserves the ABI of the original (pre-6.0) `pointee` property that
+  // used to export a getter. The current one above would export a read
+  // accessor, if it wasn't @_alwaysEmitIntoClient.
+  @_spi(SwiftStdlibLegacyABI) @available(swift, obsoleted: 1)
+  @usableFromInline
+  internal var pointee: Pointee {
+    @_transparent unsafeAddress {
+      return self
+    }
+  }
+}
 
+extension UnsafePointer where Pointee: ~Copyable {
+  /// Accesses the pointee at the specified offset from this pointer.
+  ///
+  ///
```

Comment:
```suggestion
  ///
```
Extra empty line removed